redis/src/cluster_async/mod.rs

```diff
@@ -1540,6 +1634,84 @@ impl Connect for MultiplexedConnection {
 }
 
 async fn connect_and_check<C>(
+    addr: &str,
+    params: ClusterParams,
+    socket_addr: Option<SocketAddr>,
+    conn_type: RefreshConnectionType,
+    node: Option<AsyncClusterNode<C>>,
+) -> RedisResult<AsyncClusterNode<C>>
+where
+    C: ConnectionLike + Connect + Send + Sync + 'static + Clone,
+{
+    match conn_type {
+        RefreshConnectionType::OnlyUserConnection => {
+            if let Some(node) = node {
+                let (user_conn, ip) =
+                    create_user_connection(addr, params.clone(), socket_addr).await?;
+
+                Ok(AsyncClusterNode::new(
+                    async { user_conn }.boxed().shared(),
+                    node.management_connection.clone(),
+                    ip,
+                ))
+            } else {
+                // If we do not have a relevant node for this address, it indicates that it has already been removed from the connections container.
```

Comment:
`refresh_connections` removes connections from the container, and then calls `get_or_create_conn` which calls this function. Don't wouldn't this mean that reconnections of disconnected nodes will always fail here?
If I'm right about this, we need to understand why the existing tests didn't catch this bug.

Reply:
`refresh_connections` removes a connection from the container and gets the node back, if the node was present in the connection container. We will pass the node to the get_or_create_conn. The node will be null only if the   `refresh_connections` tried to remove a node from the connection container and this node wasn't present in the container. this is the 'else' scenario we're seeing here

Reply:
I'll change it so if the node is a new node in the connection container and being added though the `refresh_connections` function, rather then in the refresh_slots function, it will force to create user & management connection, as there wasn't any connection beforehand

Reply:
Fixed